### PR TITLE
[refactor 1/n] - move utilities out of main.py to src/

### PR DESCRIPTION
**Pull request by felipemello1**
## Summary

Extract self-contained utility components from the GRPO app into the forge library for reusability.

## Changes

  Moved:
  - Episode dataclass → src/forge/rl/types.py
  - Group type alias → src/forge/rl/types.py
  - Policy type alias → src/forge/rl/types.py
  - collate() → src/forge/rl/collate.py
  - ComputeAdvantages → src/forge/rl/advantage.py
  - RewardActor → src/forge/rl/grading.py
  - drop_weights() → src/forge/util/checkpoint.py

  Deleted:
  - apps/grpo/notebook.ipynb

## Testing

Ran vs main, scores are similar

### CODE REVIEW

**Comment by JenniferWang** [comment]
I think we have an existing folder called `src/forge/data_models` -- do you plan to move them over to `rl/types`?

File: src/forge/rl/advantage.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@dataclass
+class ComputeAdvantages(ForgeActor):
+    @endpoint
+    async def compute(self, group: Group) -> list[float]:
+        # TODO: add batch processing
+        rewards = torch.tensor([[e.reward for e in group]])
+        mean = rewards.mean(1, keepdim=True)
+        std = rewards.std(1, keepdim=True)
+        advantages = (rewards - mean) / (std + 1e-4)
+        return advantages.squeeze(0).tolist()
```
Comment (JenniferWang) [inline src/forge/rl/advantage.py:26]: This being an actor does not make sense. 
Also should it be merged into `forge/util/ops.py`?

**Comment by felipemello1** [comment]
>  think we have an existing folder called src/forge/data_models -- do you plan to move them over to rl/types?

Probably. I didnt wan't this PR to be hard to review, so i didnt make changes to the functions, i just moved them. Follow up refactoring will have more changes.

File: src/forge/rl/advantage.py (hunk 1 above)
Comment (felipemello1) [reply src/forge/rl/advantage.py:26]: I dislike how loaded "util" is, but i agree that it doesnt need to be an actor. Do you think i should make the change in this PR? I am ok with it, but my intuition is that i should touch these files in follow up PRs.

Comment (JenniferWang) [reply src/forge/rl/advantage.py:26]: No. But it's generally good practice, as you go through the refactor, add a `TODO` comment so we can grip later to catch any missing refactoring.

Comment (felipemello1) [reply src/forge/rl/advantage.py:26]: will add a todo
